test/AddCombinedRowToTableTest.cpp

```diff
@@ -40,25 +49,97 @@ TEST(AddCombinedRowToTable, OneJoinColumn) {
 
     auto expected =
         makeIdTableFromVector({{7, 8, 14, 0}, {11, 10, U, U}, {14, 11, 8, 2}});
-    ASSERT_EQ(result, expected);
+    auto expectedUndefined = std::vector<size_t>{0, 0, 1, 1};
+    if (!keepJoinColumns) {
+      expected.deleteColumn(0);
+      expectedUndefined.erase(expectedUndefined.begin());
+    }
+    EXPECT_EQ(result, expected);
+
+    EXPECT_EQ(numUndefined, expectedUndefined);
```

Comment:
Instead of modifying this data, why don't you just write this as something like:
```cpp
if (keepJoinColumns) {
      EXPECT_EQ(result, makeIdTableFromVector({{7, 8, 14, 0}, {11, 10, U, U}, {14, 11, 8, 2}}));
      EXPECT_THAT(numUndefined, ::testing::ElementsAre(0, 0, 1, 1));
} else {
      EXPECT_EQ(result, makeIdTableFromVector({{8, 14, 0}, {10, U, U}, {11, 8, 2}}));
      EXPECT_THAT(numUndefined, ::testing::ElementsAre(0, 1, 1));
}
```

in my opinions this is easier to read.